src/sentry/api/event_search.py

```diff
@@ -0,0 +1,167 @@
+from __future__ import absolute_import
+
+from collections import namedtuple
+from parsimonious.grammar import Grammar, NodeVisitor
+
+from sentry.search.utils import parse_datetime_string, InvalidQuery
+
+event_search_grammar = Grammar(r"""
+# raw_search must come at the end, otherwise other
+# search_terms will be treated as a raw query
+search          = search_term* raw_search?
+search_term     = space? (time_filter / basic_filter) space?
+raw_search      = ~r".+$"
+
+# standard key:val filter
+basic_filter    = search_key sep search_value
+# filter specifically for the timestamp
+time_filter     = "timestamp" operator date_formats
+
+search_key      = ~r"[a-z]*\.?[a-z]*"
+search_value    = ~r"\S*"
+
+date_formats    = date_time_micro / date_time / date
+date            = ~r"\d{4}-\d{2}-\d{2}"
+date_time       = ~r"\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}"
+date_time_micro = ~r"\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{1,6}"
+
+# NOTE: the order in which these operators are listed matters
+# because for example, if < comes before <= it will match that
+# even if the operator is <=
+operator        = ">=" / "<=" / ">" / "<" / "=" / "!="
+sep             = ":"
+space           = ~r"\s"
+"""
+                               )
```

Comment:
You got a lot of extra whitespace here.

Reply:
this was the commit hook 😞 

but i can just remove and commit without running it bc agree it's not pretty

Reply:
lol what

Reply:
yeah idk i removed by committing without running the hook, but:
![screenshot 2018-10-26 14 39 00](https://user-images.githubusercontent.com/5026776/47593671-fb238e80-d92c-11e8-8d75-8453e7ca55f3.png)

Reply:
🍪